src/libs/Navigation/OnyxTabNavigator.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import {createMaterialTopTabNavigator} from '@react-navigation/material-top-tabs';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import Tab from '../actions/Tab';
+import ONYXKEYS from '../../ONYXKEYS';
+
+const propTypes = {
+    /* ID of the tab component to be saved in onyx */
+    id: PropTypes.string.isRequired,
+
+    /* Name of the selected tab */
+    selectedTab: PropTypes.string,
+
+    /* Children nodes */
+    children: PropTypes.node.isRequired,
+};
+
+const defaultProps = {
+    selectedTab: '',
+};
+
+// eslint-disable-next-line rulesdir/no-inline-named-export
+export const TopTab = createMaterialTopTabNavigator();
+
+// This takes all the same props as MaterialTopTabsNavigator: https://reactnavigation.org/docs/material-top-tab-navigator/#props,
+// except ID is now required, and it gets a `selectedTab` from Onyx
+function OnyxTabNavigator({id, selectedTab, children, ...rest}) {
+    return (
+        <TopTab.Navigator
+            /* eslint-disable-next-line react/jsx-props-no-spreading */
+            {...rest}
+            id={id}
+            initialRouteName={selectedTab}
```

Comment:
```suggestion
            initialRouteName={selectedTab}
            backBehavior='initialRoute'
```

Related to the reported bug https://github.com/Expensify/App/pull/23046#issuecomment-1661217402 This back behaviour is slightly better for hardware back press. The hardware back button still not having identical behaviour with the UI but it's good enough, at least for now.

Reply:
This seems correct 👍 - This is in addition to the `Navigate.dismissModal` in `MoneyRequestSelectorPage`, right?

Reply:
Yes